src/codeu/chat/client/ClientUser.java

```diff
@@ -88,27 +95,67 @@ public void showCurrent() {
     printUser(current);
   }
 
-  public void addUser(String name) {
+  //change the type to a boolean, so we can check if the user was created or not in the GUI
+  public boolean addUser(String name) {
     final boolean validInputs = isValidName(name);
 
     final User user = (validInputs) ? controller.newUser(name) : null;
 
     if (user == null) {
       System.out.format("Error: user not created - %s.\n",
-          (validInputs) ? "server failure" : "bad input value");
+          (validInputs) ? "server failure" : "bad input value"); 
     } else {
       LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
       updateUsers();
+      
+      return true; 
     }
-  }
-
+    return false;
+  }
+
+  //delete user method
+  public boolean deleteUser(String name) {
+  
+  		//get all users by name
+		Iterable <User> users = getUsers();
+		
+		User target; 
+		Uuid targetId; 
+		
+		//find user and get id
+		for(User currentUser:users){
+			if(currentUser.name.equals(name)){
+				target = currentUser; 
+				targetId = target.id;
+				
+				/*
```

Comment:
Somewhere in this method you need to call a deleteUser method on the controller like we talked about in our meeting. That method on the controller will talk to the server and communicate the changes with a new NetworkCode.